src/hotspot/share/runtime/os.cpp

```diff
@@ -1745,13 +1745,30 @@ void os::print_memory_mappings(outputStream* st) {
   os::print_memory_mappings(nullptr, (size_t)-1, st);
 }
 
+// Pretouching must use a store, not just a load.  On many OSes loads from
+// fresh memory would be satisfied from a single mapped page containing all
+// zeros.  We need to store something to each page to get them backed by
+// their own memory, which is the effect we want here.  An atomic add of
+// zero is used instead of a simple store, allowing the memory to be used
+// while pretouch is in progress, rather than requiring users of the memory
+// to wait until the entire range has been touched.  This is technically
+// a UB data race, but doesn't cause any problems for us.
 void os::pretouch_memory(void* start, void* end, size_t page_size) {
-  for (volatile char *p = (char*)start; p < (char*)end; p += page_size) {
-    // Note: this must be a store, not a load. On many OSes loads from fresh
-    // memory would be satisfied from a single mapped page containing all zeros.
-    // We need to store something to each page to get them backed by their own
-    // memory, which is the effect we want here.
-    *p = 0;
+  assert(is_power_of_2(page_size), "page size misaligned: %zu", page_size);
+  assert(page_size >= sizeof(int), "page size too small: %zu", page_size);
```

Comment:
`SIZE_FORMAT` instead of `%zu`.

Reply:
`%zu` is C99 Standard, so also available in C++14.  It's also long been supported by gcc in C++03 mode, and by at least VS2015 and later. See https://bugs.openjdk.java.net/browse/JDK-8256379.  There's been no effort to go back and change existing code, but it's been getting used in new code for a while.

Reply:
So, do you use that also for UINTX_FORMAT?

Reply:
I can't recall the last time I needed to use `INTX_FORMAT` or `UINTX_FORMAT` and had kind of forgotten about them.  But yes, those seem similarly replaceable.

Reply:
All right then, nevermind :)